test/test_mahalanobis_mixin.py

```diff
@@ -298,12 +299,18 @@ def test_components_is_2D(estimator, build_dataset):
       to_keep = np.where(np.abs(diffs.ravel()) > 1e-9)
       trunc_data = trunc_data[to_keep]
       labels = labels[to_keep]
+  if isinstance(estimator, _TripletsClassifierMixin):
+    for slice_idx in [[0, 1], [0, 2]]:
+      pairs = trunc_data[:, slice_idx, :]
+      diffs = pairs[:, 1, :] - pairs[:, 0, :]
+      to_keep = np.abs(diffs.ravel()) > 1e-9
```

Comment:
you're right, `np.where` is not needed, so I'm thinking we could remove it too for the other two cases (`_PairsClassifierMixin` and `_QuadrupletsClassifierMixin`), also, now I realize we could probably factorize the three cases together (the only thing that change is the slice(s) to take)

Reply:
Yes! I was thinking the same. It shouldn't be difficult to put them all together.